shell/imports/dates.js

```diff
@@ -0,0 +1,40 @@
+// Sandstorm - Personal Cloud Sandbox
+// Copyright (c) 2016 Sandstorm Development Group, Inc. and contributors
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+function formatFutureTime(diff) {
+  // diff is in milliseconds
+  const units = {
+    day: 86400000,
+    hour: 3600000,
+    minute: 60000,
+    second: 1000,
+  };
+
+  for (const unit in units) {
+    // If it's more than one full unit away, then we'll print in terms of this unit. This does
+    // mean that we write e.g. "1 minute" for the whole range between 2 minutes and 1 minute, but
+    // whatever, this is typical of these sorts of displays.
+    if (diff >= units[unit]) {
+      const count = Math.floor(diff / units[unit]);
```

Comment:
Let's make this Math.round(). I've actually found the problem described in the comment above is somewhat disconcerting in practice.